cli/src/vote.rs

```diff
@@ -0,0 +1,352 @@
+        vote_state.authorized_voter_pubkey
+    );
+    println!("credits: {}", vote_state.credits());
+    println!(
+        "commission: {}%",
+        f64::from(vote_state.commission) / f64::from(std::u32::MAX)
+    );
+    println!(
+        "root slot: {}",
+        match vote_state.root_slot {
+            Some(slot) => slot.to_string(),
+            None => "~".to_string(),
+        }
+    );
+    if !vote_state.votes.is_empty() {
+        println!("recent votes:");
+        for vote in &vote_state.votes {
+            println!(
+                "- slot: {}\n  confirmation count: {}",
+                vote.slot, vote.confirmation_count
+            );
+        }
+
+        // TODO: Use the real GenesisBlock from the cluster.
+        let genesis_block = solana_sdk::genesis_block::GenesisBlock::default();
+        let epoch_schedule = solana_runtime::epoch_schedule::EpochSchedule::new(
+            genesis_block.slots_per_epoch,
+            genesis_block.stakers_slot_offset,
+            genesis_block.epoch_warmup,
+        );
+
+        println!("epoch voting history:");
+        for (epoch, credits, prev_credits) in vote_state.epoch_credits() {
+            let credits_earned = credits - prev_credits;
+            let slots_in_epoch = epoch_schedule.get_slots_in_epoch(*epoch);
+            println!(
+                "- epoch: {}\n  slots in epoch: {}\n  credits earned: {}",
+                epoch, slots_in_epoch, credits_earned,
+            );
+        }
+    }
+    Ok("".to_string())
+}
+
+pub fn process_uptime(
+    rpc_client: &RpcClient,
+    _config: &WalletConfig,
+    vote_account_pubkey: &Pubkey,
+    aggregate: bool,
+    span: Option<u64>,
+) -> ProcessResult {
+    let vote_account = rpc_client.get_account(vote_account_pubkey)?;
+
+    if vote_account.owner != solana_vote_api::id() {
+        Err(WalletError::RpcRequestError(
+            format!("{:?} is not a vote account", vote_account_pubkey).to_string(),
+        ))?;
+    }
+
+    let vote_state = VoteState::deserialize(&vote_account.data).map_err(|_| {
+        WalletError::RpcRequestError(
+            "Account data could not be deserialized to vote state".to_string(),
+        )
+    })?;
+
+    println!("Node id: {}", vote_state.node_pubkey);
+    println!(
+        "Authorized voter pubkey: {}",
+        vote_state.authorized_voter_pubkey
+    );
+    if !vote_state.votes.is_empty() {
+        println!("Uptime:");
+
+        // TODO: Use the real GenesisBlock from the cluster.
+        let genesis_block = solana_sdk::genesis_block::GenesisBlock::default();
+        let epoch_schedule = solana_runtime::epoch_schedule::EpochSchedule::new(
+            genesis_block.slots_per_epoch,
+            genesis_block.stakers_slot_offset,
+            genesis_block.epoch_warmup,
+        );
+
+        let epoch_credits_vec: Vec<(u64, u64, u64)> = vote_state.epoch_credits().copied().collect();
+
+        let epoch_credits = if let Some(x) = span {
```

Comment:
Should we warn the user if span is longer than the available epochs?  I don't think that needs to be an error though